greenpeace/planet4-master-theme
# PLANET-7044 Create new toggle that enables all new IA features

Ref: https://jira.greenpeace.org/browse/PLANET-7044

---

- Adds a new checkbox in _Planet 4 > Navigation_ and completely removing the whole _Planet 4 > Information Architecture_ section.
- Adds a new migration that set this checkbox to `true` if any of the deprecated feature flags was active.


### Testing
1. Go to Admin Panel: _Planet 4 > Information Architecture_ and enable at least one of the features.
2. Get inside the php container: `make php-shell`.
3. Run migration script: `wp p4-run-activator`.
4. Go to Admin Panel: _Planet 4 > Navigation_. The "New Information Architecture" checkbox should be enabled.
5. Check all the relevant features work: 
   - Actions post type is active
   - Default Action slug setting is visible in Permalinks settings
   - Auto-generated listing pages use pagination
   - Posts have their Category link above title
   - Mobile tabs menu works in the frontend
6. Disable the "New Information Architecture" checkbox and all the above featues should be disabled.


### Acceptance tests
Acceptance tests are failing in this branch because of [this line](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/d032936b703d998ee7560ba3ec058836ffa62b39/classes/blocks/class-covers.php#L11) in the blocks plugin. Created the linked PR in plugins that uses this commit as a submodule to verify that tests will pass when both PRs get merged.
